tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +137,139 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
 	await expect(page).toHaveURL(`/onboarding`)
 })
 
+test('onboarding with GitHub OAuth', async ({ page }) => {
+	// grab mock github users from tests/fixtures/github
+	const ghUsers = await getGitHubUsers()
+	const ghUser = ghUsers.find(u => u.code === MOCK_CODE_GITHUB)
+
+	// a github user can have numerous emails, but only one is primary,
+	// so we use it to query our users relation: if a user with this email
+	// is there, we simply create a link between the 'user' and  'connection'
+	// entities and make a new session, since the user is already on board;
+	// this behavior will be tested later on, but for now we need to prepare the ground:
+	const user = await prisma.user.findUnique({
+		where: { email: ghUser?.primaryEmail.toLowerCase() },
+	})
+	if (user) await prisma.user.delete({ where: { id: user.id } })
+
+	// we remove the connection if any to keep the test clean
+	const conn = await prisma.connection.findFirst({
+		where: { providerName: 'github', userId: user?.id },
+	})
+	if (conn) await prisma.connection.delete({ where: { id: conn.id } })
+
+	await page.goto('/signup')
+	await page.getByRole('button', { name: /signup with github/i }).click()
+
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+	expect(
+		page.getByText(new RegExp(`welcome aboard ${ghUser?.primaryEmail}`, 'i')),
+	).toBeVisible()
+
+	// fields are pre-populated for the user
+	const usernameInput = page.getByRole('textbox', { name: /username/i })
+	expect(usernameInput).toHaveValue(
+		// @ts-ignore
+		normalizeUsername(ghUser!.profile.login),
+	)
+	expect(page.getByRole('textbox', { name: /^name/i })).toHaveValue(
+		ghUser!.profile.name,
+	)
+
+	// button currently in 'idle' (neutral) state and so has got no companion
+	const createAccountButton = page.getByRole('button', {
+		name: /create an account/i,
+	})
+	expect(createAccountButton.getByRole('status')).not.toBeVisible()
+
+	// attempt 1
+	// Username is a string of alphanums and underscores from 3 to 20 chars
+	// long, that (NB!) will be lowercased upon sumbission. See: app/utils/user-validation.ts:
```

Comment:
What does "NB!" mean?

Reply:
Means ["pay attention!"](https://en.wikipedia.org/wiki/Nota_bene), but those sneakers make made it confusing :smile: